x-pack/plugins/telemetry_collection_xpack/server/index.ts

```diff
@@ -5,8 +5,6 @@
  * 2.0.
  */
 
-export type { ESLicense } from './telemetry_collection';
```

Comment:
Is this something developers should be able to access even if not in use rn?

Reply:
Ideally, this exported type should be used from within the `telemetry_collection_xpack` only.
We've removed the export at this level to make it a bit more hard for other plugins to access it.
We've kept the export in `./telemetry_collection/index.ts`:

```
import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';

export type ESLicense = estypes.LicenseGetLicenseInformation;
```

Other plugins should use a similar strategy if they need to use this interface, or perhaps we can consider exporting it from one of the `platform/shared` plugins.